src/devices/machine/mb8421.h

```diff
@@ -72,90 +77,153 @@
 //  TYPE DEFINITIONS
 //**************************************************************************
 
-// ======================> mb8421_device
+// ======================> dual_port_mailbox_ram_base
 
-class mb8421_master_device : public device_t
+template <typename Type>
```

Comment:
Please make the address bits and data bits template parameters so masking doesn't require runtime member fetches and can use immediate operand instructions:
* This line should become `template <typename Type, unsigned AddrBits, unsigned DataBits>`
* Remove the members for the masks and mailbox addresses
* When you need the address mask, use `make_bitmask<offs_t>(AddrBits)`
* When you need the data mask, use `make_bitmask<Type>(DataBits)`
* Use either a `static constexpr` helper for each mailbox address or just use the expression to calculate it when you need it

(`make_bitmask` is a `constexpr` helper from src/lib/util/coretmpl.h)